### ISSUE TITLE
(16,43): error TS2305: ...node_modules/@types/react/index"' has no exported member 'ReactPortal'.

### ISSUE THREAD

**Issue by jahoops** [wmonk/create-react-app-typescript#188]
Yes, I tried everything.  It is quite simple, I follow the directions exactly and create my-app.  That's all.  I get the message:
(16,43): error TS2305: Module '"/home/mint/source/repos/my-app/node_modules/@types/react/index"' has no exported member 'ReactPortal'.

I look and that is correct, there is no ReactPortal.  I read the instructions, and downgrade to npm 4.6, and start over.  Same message.  No matter what I do, same message.

It's probably me, being new to TypeScript, but I was just going to try out TypeScript because all the different versions of JavaScript were driving me crazy.  It's looking like dealing with TypeScript is no better.  I just don't have time for this.

**Comment by peterblazejewicz** [comment]
@jahoops The updated @types/react is not yet there on NPM, hence the problem. You could use augmented TypeScript module declaration to fix a problem with 'create-react-app-typescript' output. I'm working on template project and here is a fix I've used just now:
https://github.com/peterblazejewicz/the-ui-library/commit/4d3c570d14c92b54ce7f97b04aa0a40a1f171fdc
Good luck!

**Comment by jahoops** [comment]
Wow!  Very impressive and quick fix!  I just created a types folder, added the react.d.ts, and made the changes to tsconfig.json and now it is working!

Thank you very much.  I am trying to wedge some React components into an existing app.  Since I coded in C# for a long time I thought TypeScript might make my life easier, because all the different versions, styles, transpilers, shims, etc. for React and JavaScript are driving me crazy.
